compiler/rustc_infer/src/infer/outlives/verify.rs

```diff
@@ -56,6 +68,30 @@ impl<'cx, 'tcx> VerifyBoundCx<'cx, 'tcx> {
             }
         }
 
+        // A parameter or placeholder bound can also be aliased to an associated type
+        // so we need to inspect in the "reverse"
+        for &predicate in self.projection_predicates {
+            let Some(ty::ProjectionPredicate { term, projection_ty }) = predicate.no_bound_vars()
+            else {
+                continue;
```

Comment:
... so if we are here, we have, for now, `T == for<'a, 'b, ...> <... as ...>::...`.

I guess it is safe to say that `T: 'static`?

Reply:
I retract. It is not necessarily.